### PR TITLE
Move DAPR GRPC port inside client

### PR DESCRIPTION
**Pull request by dwhiteddsoft**
Other DAPR SDKs "hide" the obtaining of the GRPC port from the user of client. This change is simply made in that spirit. Changes were:
- a modification to the client.rs file to obtain the environment variable there and parse
- a modification to the error.rs to surface errors from this
- a modification to the pub/sub example to reflect this change

### CODE REVIEW

**Comment by NickLarsenNZ** [comment]
@yaron2 is there anything holding this back?

**Comment by NickLarsenNZ** [comment]
or is @shubham1172 able to approve?

**Comment by NickLarsenNZ** [comment]
@yaron2 I see you merged another PR recently. Are you able to take a look at this at the same time?

File: src/client.rs
Comment (shubham1172) [inline src/client.rs:19]: Can we also support `port: Option<String>` that overrides the GRPC port environment variable?
Also, if the environment variable does not exist, it can fallback to a standard default, like `50051`.

File: src/client.rs
Comment (shubham1172) [inline src/client.rs:21]: It's better to use `match` and handle the errors here explicitly (we can revert error.rs then). In case of error parsing the environment variable, we can log and fallback to a standard default.

**Comment by NickLarsenNZ** [comment]
@dwhiteddsoft are you still interested in making the suggested changes?
Or @shubham1172 can we approve and get it in, then your suggestions could be individual issues with implementations coming later?

**Comment by shubham1172** [comment]
> @dwhiteddsoft are you still interested in making the suggested changes? Or @shubham1172 can we approve and get it in, then your suggestions could be individual issues with implementations coming later?

Yes, those changes can be a follow up IMO. @dapr/maintainers-rust-sdk to have a look and merge.

**Review by shubham1172 — APPROVED** [review]

**Comment by dwhiteddsoft** [comment]
Just got back from being out. It was so long ago, I thought it wasn't maintained any longer. If I can help in any way, please let me know

**Comment by mikeee** [comment]
@dwhiteddsoft Sorry it's been so long since this has been picked up, would you mind if I resolve conflicts on your behalf and get this merged? I'm in agreement that the comments can be picked up at a later date

**Comment by dwhiteddsoft** [comment]
> @dwhiteddsoft Sorry it's been so long since this has been picked up, would you mind if I resolve conflicts on your behalf and get this merged? I'm in agreement that the comments can be picked up at a later date

Sure. If there is anything I can or should do please let me know

**Comment by mikeee** [comment]
@dwhiteddsoft sorry - just one thing I'll need your help with please, looks like the DCO is failing as the commit wasn't signed off:
https://github.com/dapr/rust-sdk/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

**Comment by dwhiteddsoft** [comment]
> @dwhiteddsoft sorry - just one thing I'll need your help with please, looks like the DCO is failing as the commit wasn't signed off: https://github.com/dapr/rust-sdk/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

ack - will work on tonight

**Comment by dwhiteddsoft** [comment]
> @dwhiteddsoft sorry - just one thing I'll need your help with please, looks like the DCO is failing as the commit wasn't signed off: https://github.com/dapr/rust-sdk/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

I have done as the docs asked. I have never done this particular step before so I hope I did it correctly.

**Comment by mikeee** [comment]
I can't see a force push to the branch, could you please try again?

**Comment by mikeee** [comment]
> > I can't see a force push to the branch, could you please try again?
> 
> I think I did it correctly. You can reach out to me directly if a call would be easier

https://github.com/dapr/rust-sdk/pull/54/checks?check_run_id=23110606554

This run has a set of instructions (slightly different) to the contributor docs. Give this a shot as it should work too, if the next force push doesn't work feel free to shoot me an email to the one on my profile 👍

**Comment by mikeee** [comment]
I'll set to pass manually 👍

**Comment by dwhiteddsoft** [comment]
> I'll set to pass manually 👍

sorry

**Review by mikeee — APPROVED** [review]

**Pull request merged**
